packages/block-library/src/query-loop/edit.js

```diff
@@ -78,6 +79,29 @@ export default function QueryLoopEdit( {
 			if ( sticky ) {
 				query.sticky = sticky === 'only';
 			}
+
+			// When you insert this block outside of the edit site then store
+			// does not exist therefore we check for its existence.
+			if ( isGlobalQuery && select( 'core/edit-site' ) ) {
```

Comment:
Now that I better understand the proposed changes, it looks like we can avoid passing `isGlobalQuery` down, and we could modify the context passed in the Query block instead. What would be even better is to read the `slug` and other query params from the context set by the editor. In fact, there might be something like that already:

https://github.com/WordPress/gutenberg/blob/5c530b2a9de44381c9add6ad23967fcd36042bc7/packages/edit-site/src/components/editor/index.js#L160-L183

I'm going to investigate how it all works as it would make everything so much easier to control for all blocks like Query Pagination, or for example Calendar block that should look different depending on query args selected.

Reply:
If we are going this way (not decoupling context with preview), that was the direction I was following as well in my explorations. `setPage` sets context to blocks that needed and was setting info in FSE navigation sidebar.

Reply:
@aristath, this is the only part of this PR that I'm not particularly happy about but at the same time it allows us to move forward so I think it's an acceptable compromise. I plan to look into the context propagation separately anyway.